lib/membrane_http_adaptive_stream/sink_bin.ex

```diff
@@ -198,17 +207,33 @@ defmodule Membrane.HTTPAdaptiveStream.SinkBin do
   end
 
   @impl true
-  def handle_element_end_of_stream({:sink, _}, _ctx, %{streams_counter: 1} = state) do
-    state = Map.update!(state, :streams_counter, &(&1 - 1))
-    {{:ok, notify: :end_of_stream}, state}
+  def handle_element_start_of_stream(
+        {:sink, _},
+        _ctx,
+        %{streams_to_start: 1, started?: false} = state
+      ) do
+    {{:ok, notify: :start_of_stream}, %{state | streams_to_start: 0, started?: true}}
   end
 
   @impl true
-  def handle_element_end_of_stream({:sink, _}, _ctx, state) do
-    state = Map.update!(state, :streams_counter, &(&1 - 1))
+  def handle_element_start_of_stream({:sink, _}, _ctx, state) do
+    {:ok, Map.update!(state, :streams_to_start, &(&1 - 1))}
+  end
+
+  @impl true
+  def handle_element_start_of_stream(_element, _ctx, state) do
     {:ok, state}
   end
 
+  def handle_element_end_of_stream({:sink, _}, _ctx, %{streams_to_end: 1, ended?: false} = state) do
+    {{:ok, notify: :end_of_stream}, %{state | streams_to_end: 0, ended?: true}}
+  end
```

Comment:
What `ended?: false` really gives us here? I think it will make sure that we don't send :end_of_stream twice, but I think it would be an expected behavior if someone connected a new pad after processing finished.

This obviously applies to `started?: false` as well

Reply:
Yup, forgot to change that after our internal discussion.